.github/haskell.yml

```diff
@@ -0,0 +1,69 @@
+name: Haskell CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install Haskell build environment
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install software-properties-common
+        sudo add-apt-repository ppa:hvr/ghc
+
+        sudo apt-get -y install locales curl sudo git
+        sudo apt-get -y install g++ libstdc++6
+        sudo apt-get -y install libsnappy1v5 libtinfo-dev libsnappy-dev jq
+        sudo apt-get -y install zlib1g zlib1g-dev
+        sudo apt-get -y install libz3-dev
+        sudo apt-get -y install libpq5 libpq-dev
+        sudo apt-get -y install alex-3.1.7 cabal-install-3.0 happy-1.19.5
+        sudo apt-get -y install hpack hspec-discover libyaml-0-2
+        sudo apt-get -y install ghc-8.8.1
+        sudo apt-get -y install zip
+        sudo apt-get -y install wget
+
+        sudo apt-get -y remove --purge software-properties-common
+        sudo apt-get -y autoremove
+
+        sudo ln -s /opt/cabal/bin/cabal /usr/local/bin/cabal
+        sudo ln -s /opt/ghc/bin/ghc /usr/local/bin/ghc
+        sudo ln -s /opt/ghc/bin/ghc-pkg /usr/local/bin/ghc-pkg
+        cabal --version
+        ghc --version
+
+    - name: Install cabal-cache tool
+      run: |
+        CACHE_TOOL_TAG=$(curl -s -X GET https://api.github.com/repos/haskell-works/cabal-cache/releases/latest | jq -rc .tag_name)
+        echo "Downloading: https://github.com/haskell-works/cabal-cache/releases/download/${CACHE_TOOL_TAG}/cabal-cache_x86_64_linux.tar.gz"
```

Comment:
`cabal-cache` is a CI optimisation.  It caches libraries that have been built before to avoid needing to build them again.

`cabal-cache` doesn't actually end up as a dependency and you could remove it at the cost of build time.

The build time in CI with `cabal-cache` support is about 1 minute: https://circleci.com/gh/haskell-works/r-tree/26#build-timing/containers/0

If you're interested in setting up your own cache, it only requires an AWS S3 bucket, which are relatively inexpensive and enables sharing of the cache between multiple projects.

At the moment, your library has few enough dependencies that without the cache the build time is only about 1 minute more, so it's probably fine without.  I'll look at doing a cacheless version for now.